File: solutions/ProblemSetsEasy/solve/ruby/ProblemSetsEasySolver.rb
#! /usr/bin/env ruby

require_relative "ProblemSetsEasy"

require "gettc/types"
include Gettc

def init()
    gettc_home = File.join(ENV["HOME"], ".gettc")
    if ENV.has_key?("GETTC_HOME")
        gettc_home = ENV["GETTC_HOME"]
    end 
    gettc_home = File.expand_path(gettc_home)
    $LOAD_PATH << File.join(gettc_home, "include", "ruby")
    require "topcoder"
    include TopCoder
end

def main()
    reader = Reader.new(IO.read(ARGV[0]))
    E = reader.next(TInt)
    reader.next()
    EM = reader.next(TInt)
    reader.next()
    M = reader.next(TInt)
    reader.next()
    MH = reader.next(TInt)
    reader.next()
    H = reader.next(TInt)

    result = ProblemSetsEasy.new().maxSets(E, EM, M, MH, H)
    IO.write(ARGV[1], Writer.new().next(result, TInt).to_s())
end
init()
main()

File: solutions/ProblemSetsEasy/solve/python/ProblemSetsEasySolver.py
#! /usr/bin/env python
import os, sys
import ProblemSetsEasy

def init():
    default_path = os.path.join(os.getenv("HOME"), ".gettc")
    gettc_home = os.path.abspath(os.getenv("GETTC_HOME", default_path))
    include_dir = os.path.join(gettc_home, "include/python")
    sys.path.append(include_dir)

def main():
    import topcoder as tc
    with open(sys.argv[1], "r") as fi:
        input = fi.read()
        reader = tc.Reader(input)
    
        E = reader.next("int")
        reader.next()
        EM = reader.next("int")
        reader.next()
        M = reader.next("int")
        reader.next()
        MH = reader.next("int")
        reader.next()
        H = reader.next("int")

    result = ProblemSetsEasy.maxSets(E, EM, M, MH, H)
    with open(sys.argv[2], "w") as fo:
        fo.write(tc.write(result, "int"))

if __name__ == "__main__":
    init()
    main()
